### PR TITLE
[werft/observability] Enable tracing with open-telemetry collector

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

As one of our first experiments with tracing with honeycomb, we want to push traces from Gitpod components to Honeycomb and learn how to work with it.

OpenTelemetry configuration is being added to monitoring-satellite, behind the external variable `is_preview`. More details in the [development branch](https://github.com/gitpod-io/observability/pull/18).

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/195

## How to test

Comment below:
```
/werft run with-observability
```
You can spin up a workspace from this branch and notice how OpenTelemetry collector's pod is configured and its logs or you could also search for the traces in [Honeycomb's dataset](https://ui.honeycomb.io/gitpod/datasets/preview-environments).

## Release Notes

```release-note
NONE
```

### Note worthy comments

We probably want to add some [processing pipelines](https://opentelemetry.io/docs/collector/configuration/#processors) to the collector that adds the branch name that is emitting these traces as an attribute. 

I'm hoping that this could be done in a follow-up PR.

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **arthursens** and additionally assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by mads-hartmann — COMMENTED** [review]
@ArthurSens if we merge this we will no longer get traces into Jaeger for preview environments rights?

Ideally we'd have Jaeger and Honeycomb live side-by-side for a while, especially in production, so people can migrate to Honeycomb gradually. That also allows us to deploy Honeycomb at any time without having to do any big-bang release or company-wide coordination.

So before merging this PR, could you add a follow up PR to https://github.com/gitpod-io/observability/pull/18 which uses the [Jaeger Exporter](https://opentelemetry-python.readthedocs.io/en/latest/exporter/jaeger/jaeger.html) to have the otel-collector send traces to Jaeger?

**Comment by ArthurSens** [comment]
> @ArthurSens if we merge this we will no longer get traces into Jaeger for preview environments rights?

That is true, but I think our Jaeger installation is already broken in previews anyway 🤷 

> So before merging this PR, could you add a follow up PR to [gitpod-io/observability#18](https://github.com/gitpod-io/observability/pull/18) which uses the [Jaeger Exporter](https://opentelemetry-python.readthedocs.io/en/latest/exporter/jaeger/jaeger.html) to have the otel-collector send traces to Jaeger?

Absolutely! But I don't think an extra component is needed here. The whole point of OpenTelemetry is to make it easier to send telemetry data to multiple different vendors, including OSS projects.

All we need is to add an extra exporter into the collector's configmap

**Comment by mads-hartmann** [comment]
>Absolutely! But I don't think an extra component is needed here. The whole point of OpenTelemetry is to make it easier to send telemetry data to multiple different vendors, including OSS projects.
> All we need is to add an extra exporter into the collector's configmap that redirect traces to Jaeger

I know 😁 I meant to link to how to configure the otel collector to send traces to Jaeger but see that I linked to a Python module - sorry about that 😂

If you can get a quick 👍  from the teams to stop sending traces to Jaeger in preview environments then go ahead and merge this PR. Then we'll just test having the otel collector send traces to Honeycomb + Jaeger in staging instead of preview environments.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **arthursens** and additionally assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#195*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by mads-hartmann — APPROVED** [review]
Looks good to me 🚀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: f9cdf652906bea39d9b9e2824625b5f175f3a1db</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6634#pullrequestreview-808292008" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#195*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [mads-hartmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6634#issuecomment-971368151" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#195*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [mads-hartmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6634#issuecomment-971368151" title="Approved">mads-hartmann</a>*

Associated issue: *#195*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [mads-hartmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
